src/main.c

```diff
@@ -357,6 +365,12 @@ static void focal_create_main_window(GApplication* app, FocalApp* fm)
 	gtk_widget_show_all(fm->mainWindow);
 
 	app_header_set_event(FOCAL_APP_HEADER(fm->header), NULL);
+
+	fm->sync_timer_id = g_timeout_add_seconds(
+		120 /* sync interval, seconds */,
+		G_SOURCE_FUNC(sync_func),
+		fm);
+	g_source_set_name_by_id(fm->sync_timer_id, "[focal] sync_timer");
```

Comment:
Just curious because I haven't used this in other `g_timeout_add` calls, what is this useful for?

Reply:
Not entirely sure, but example I was looking at used it and it seemed potentially useful.  Maybe not-- I was still piecing together what a source was, and so I was hardly in a position to snub some debuggability when it was right there :).  I did, however, imagine it might end up as the thread name, but that doesn't appear to be the case.  Which is good because spawning polling thread for each source does seem a bit silly.  Anyway, since you asked I and because I was curious myself, here's what the docs[1] say:

> Sets the name of a source using its ID.
> 
> This is a convenience utility to set source names from the return value of g_idle_add(), g_timeout_add(), etc.
> 
> It is a programmer error to attempt to set the name of a non-existent source.
> 
> More specifically: source IDs can be reissued after a source has been destroyed and therefore it is never valid to use this function with a source ID which may have already been removed. An example is when scheduling an idle to run in another thread with g_idle_add(): the idle may already have run and been removed by the time this function is called on its (now invalid) source ID. This source ID may have been reissued, leading to the operation being performed against the wrong source.
> 
> 

`g_source_set_name` adds:

> 
> Sets a name for the source, used in debugging and profiling. The name defaults to NULL.
> 
> The source name should describe in a human-readable way what the source does. For example, "X11 event queue" or "GTK+ repaint idle handler" or whatever it is.
> 
> It is permitted to call this function multiple times, but is not recommended due to the potential performance impact. For example, one could change the name in the "check" function of a GSourceFuncs to include details like the event type in the source name.
> 
> Use caution if changing the name while another thread may be accessing it with g_source_get_name(); that function does not copy the value, and changing the value will free it while the other thread may be attempting to use it.

I'm still slightly unclear what, concretely, might access and display this name.  Maybe some of glib/gtk logging or assertion macros use this as a bit of execution context info?

I am happy with whatever your preference is here, maybe later I'll chase down a better answer as to who might see this name, or said another way: "why is this not essentially a comment with runtime cost" :laughing: 

[1] https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#g-source-set-name-by-id